chore: update linters

rodrodros opened the pull request.

rodrodros committed:
chore: update linters

thiagodeev commented on .github/workflows/linter.yml:
```diff
@@ -14,12 +14,20 @@ jobs:
         with:
-          version: v2.4.0
\ No newline at end of file
+          version: latest
```
I disagree with this change.

Arguments:
- We will be susceptible to a supply chain attack if a future version of the linter is compromised, since our CI will always get the latest release, even though it's compromised.
- Incompatible new versions could break our workflow. E.g.: when golangci v3.0.0 is released, we'll fetch it and break our CI

It's a good practice to use fixed versions and update after

WDYT?

rodrodros replied:
1. The attack comes from using `golangci/golangci-lines-action@v8`, not from setting `latest`. I think it creates more hassle and hurts readability. Take Copilot comments with a pinch of salt.
2. What's the big deal with out CI breaking when golangci v3.0.0 releases? We then make the decision to update to v3.0.0 or set a different one.

thiagodeev committed:
chore: fix lll linter errors in the typeddata pkg

Start to implement the new rules in the lint diff file

thiagodeev committed:
fix: fix lll linter errors in the utils pkg

thiagodeev committed:
fix: fix lll linter errors in the internal pkg

thiagodeev committed:
fix: new lll linter errors in the rpc pkg

thiagodeev committed:
fix: new lll linter errors in the account pkg

thiagodeev committed:
fix: new lll linter errors in all other small pkgs

thiagodeev committed:
feat: apply new nolintlint lint rules and implement them in the code

thiagodeev committed:
feat: add new staticcheck linter rule and update part of the code with it

- Fix part of the linter errors triggered by it. Others will be fixed in a next commit
- Updated changelog with the changes

thiagodeev committed:
refactor: update interface assertions to use nil pointers

thiagodeev committed:
refactor: rename typedData package to typedata and update related references

- Updated changelog to reflect these changes.

thiagodeev committed:
refactor: more changes related to the ST1003 staticcheck linter

thiagodeev committed:
refactor: more changes related to the ST1003 staticcheck linter 2

thiagodeev committed:
refactor: more changes related to the ST1003 staticcheck linter 3

thiagodeev committed:
refactor: more changes related to the ST1003 staticcheck linter 4

thiagodeev committed:
refactor: fixing other staticcheck linter errors

thiagodeev committed:
refactor: enable all gocritic linters

thiagodeev committed:
refactor: allow gosec linter and change code to support it

thiagodeev committed:
ci: enable all govet linters except fieldalignment

thiagodeev committed:
ci: remove more exclusions from the linter file

thiagodeev committed:
Merge branch 'main' into rdr/update-linters

thiagodeev committed:
ci: remove golangci_diff.yaml and update linter workflow to use specific versions

thiagodeev committed:
chore: update golangci-lint version to v2.5.0 and fix linter
